### PR TITLE
Measure processing time of model responses

### LINKED ISSUE #105: Measure Model response time
Measure how long it takes a model to respond to a query. Unit: milliseconds.

- [x] add new assessment to measure model response time
- [x] make sure it is excluded from score computation
- [x] make sure all our CSV stuff works as expected

### PR DESCRIPTION
**Pull request by bauersimon**
Closes #105

### CODE REVIEW

**Comment by bauersimon** [comment]
Example result with our free models from openrouter.

```csv
model,language,repository,score,coverage-statement,files-executed,processing-time,response-no-error,response-no-excess,response-with-code
openrouter/google/gemma-7b-it:free,golang,golang/plain,3,0,0,2519,1,1,1
openrouter/google/gemma-7b-it:free,java,java/plain,14,10,1,1982,1,1,1
openrouter/gryphe/mythomist-7b:free,golang,golang/plain,1,0,0,7099,1,0,0
openrouter/gryphe/mythomist-7b:free,java,java/plain,2,0,0,9565,1,0,1
openrouter/mistralai/mistral-7b-instruct:free,golang,golang/plain,1,0,0,1955,1,0,0
openrouter/mistralai/mistral-7b-instruct:free,java,java/plain,14,10,1,2472,1,1,1
openrouter/nousresearch/nous-capybara-7b:free,golang,golang/plain,1,0,0,3731,1,0,0
openrouter/nousresearch/nous-capybara-7b:free,java,java/plain,14,10,1,2826,1,1,1
openrouter/openrouter/cinematika-7b:free,golang,golang/plain,2,0,0,1661,1,0,1
openrouter/openrouter/cinematika-7b:free,java,java/plain,13,10,1,2247,1,0,1
openrouter/undi95/toppy-m-7b:free,golang,golang/plain,1,0,0,3041,1,0,0
openrouter/undi95/toppy-m-7b:free,java,java/plain,2,0,0,3576,1,0,1
symflower/symbolic-execution,golang,golang/plain,14,10,1,206,1,1,1
symflower/symbolic-execution,java,java/plain,14,10,1,3028,1,1,1
```

**Comment by bauersimon** [comment]
Actually... let me investigate why symflower is fast for Go and slow for Java...

**Comment by bauersimon** [comment]
Aye... can confirm running manually that symflower is slower for Java

File: evaluate/metrics/assessment.go [hunk 1]
```diff
@@ -99,8 +100,10 @@ func (a Assessments) Score() (score uint) {
 		return 0
 	}
 
-	for _, value := range maps.Values(a) {
-		score += value
+	for key, value := range a {
+		if pointsPerAssessment[key] != 0 {
```
Comment (Munsio) [inline evaluate/metrics/assessment.go:104]: Why the change? The value is 0 for `processing-time` so it shouldn't really matter?

File: evaluate/repository.go [hunk 2]
```diff
@@ -60,6 +60,9 @@ func Repository(logger *log.Logger, resultPath string, model evalmodel.Model, la
 
 			continue
 		}
+		if assessments[metrics.AssessmentKeyProcessingTime] == 0 {
+			return nil, nil, pkgerrors.Errorf("no model response time measurement present for %q at repository %q", model.ID(), repositoryPath)
```
Comment (Munsio) [inline evaluate/repository.go:64]: Shouldn't this be handled the same way the `err` response from `model.GenerateTestsForFile` is handled because as soon as one file does not work with the model it will break the whole execution instead of marking that particular file as "errored"

**Review by Munsio — CHANGES_REQUESTED** [review]

File: evaluate/metrics/assessment.go (hunk 1 above)
Comment (bauersimon) [reply evaluate/metrics/assessment.go:104]: We don't want to give models a higher score because they take longer to answer.

File: evaluate/repository.go (hunk 2 above)
Comment (bauersimon) [reply evaluate/repository.go:64]: Hmmm. Well... if there is no response time measurement it's definitely our fault not the models. So that means we forgot to implement the timing measurement for the model. And I think that justifies a hard error.

Comment (bauersimon) [reply evaluate/repository.go:64]: Just to ensure we can never forget to implement the timing when we add new models.

**Review by Munsio — APPROVED** [review]

File: cmd/eval-dev-quality/cmd/evaluate_test.go [hunk 3]
```diff
@@ -126,7 +127,7 @@ func TestEvaluateExecute(t *testing.T) {
 			},
 
 			ExpectedOutputValidate: func(t *testing.T, output string, resultPath string) {
-				assert.Contains(t, output, `Evaluation score for "symflower/symbolic-execution" ("code-no-excess"): score=14, coverage-statement=10, files-executed=1, response-no-error=1, response-no-excess=1, response-with-code=1`)
+				assert.Regexp(t, `.*Evaluation score for "symflower/symbolic-execution" \("code-no-excess"\): score=14, coverage-statement=10, files-executed=1, processing-time=\d+, response-no-error=1, response-no-excess=1, response-with-code=1.*`, output)
```
Comment (zimmski) [inline cmd/eval-dev-quality/cmd/evaluate_test.go:130]: why .* at the end? what should come after that? Isn't that the only line?

Comment (zimmski) [inline cmd/eval-dev-quality/cmd/evaluate_test.go:130]: Since we are testing with a real execution, please make sure that procsessing time is > 0

Comment (zimmski) [reply cmd/eval-dev-quality/cmd/evaluate_test.go:130]: And start.

Comment (bauersimon) [reply cmd/eval-dev-quality/cmd/evaluate_test.go:130]: Was able to remove both. Guess it's just about that the match exists, not that it needs to be a complete match.

**Comment by zimmski** [comment]
Taking this PR.
